### PR TITLE
feat: Move proto to external repository

### PR DESCRIPTION
**Pull request by yevgenypats**
This removes protobuf files and generated code to https://github.com/cloudquery/plugin-pb/ and https://github.com/cloudquery/plugin-pb-go/.

This will make it possible for the CLI not import the plugin-sdk (any other clients) as well as decouple version upgrade from protobuf updates and SDK updates.

This is ready for first review but Im still working on a PR for the CLI to test that it all works properly before merging.

### CODE REVIEW

File: serve/source_v1_test.go [hunk 1]
```diff
@@ -255,31 +270,35 @@ func TestSourceFail(t *testing.T) {
...
-		t.Fatal("expected error but got nil")
+	for {
+		_, err := syncClient.Recv()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			break
+		}
+		t.Fatalf("expected error got nul")
```
Comment (hermanschaaf) [inline serve/source_v1_test.go:301]:
```suggestion
		t.Fatalf("expected error but got nil")
```

File: serve/source_v1_test.go [hunk 2]
```diff
@@ -104,77 +106,85 @@ func TestSourceSuccess(t *testing.T) {
 	if err != nil {
 		t.Fatalf("Failed to dial bufnet: %v", err)
 	}
-	c, err := clients.NewClient(ctx, specs.RegistryGrpc, "", "", clients.WithGRPCConnection(conn), clients.WithNoSentry())
```
Comment (hermanschaaf) [inline serve/source_v1_test.go:107]: How does the `--no-sentry` option get passed to the plugins now? 🤔

File: serve/source_v0_test.go [hunk 3]
```diff
@@ -56,72 +56,81 @@ func TestSourceSuccessV1(t *testing.T) {
...
+			break
+		}
+		if err != nil {
+			t.Fatal(err)
+		}
 		var resource schema.DestinationResource
-		if err := json.Unmarshal(resourceB, &resource); err != nil {
-			t.Fatalf("failed to unmarshal resource: %v", err)
+		if err := json.Unmarshal(r.Resource, &resource); err != nil {
+			t.Fatal(err)
```
Comment (hermanschaaf) [inline serve/source_v0_test.go:120]:
```suggestion
			t.Fatal("failed to unmarshal resource: %v", err)
```

File: serve/source_v1_test.go (hunk 2 above)
Comment (hermanschaaf) [reply serve/source_v1_test.go:107]: I guess the code for managing the plugins will now live in the CLI?

Comment (yevgenypats) [reply serve/source_v1_test.go:107]: Yes.

File: serve/source_v0_test.go (hunk 3 above)
Comment (yevgenypats) [reply serve/source_v0_test.go:120]: because it's just test then I prefer to to add formatting because it printing the line anyway.

Comment (yevgenypats) [reply serve/source_v0_test.go:120]: some places had formatting and some didn't so I tried to make it consistent without formatting in test and just call `t.Fatal`

File: serve/source_v1_test.go (hunk 2 above)
Comment (yevgenypats) [reply serve/source_v1_test.go:107]: For the test because it's already running sentry is not relevant here.

**Review by hermanschaaf — APPROVED** [review]
Haven't tested it myself yet, but looks good from a read-through. Nothing too scary :)

**Pull request merged**